feat(curriculum): add second flexbox workshop to FSD cert

sebacodes opened the pull request:
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitHub Codespaces.

Closes freeCodeCamp/CurriculumExpansion#775

sebacodes committed:
add second flexbox workshop

sebacodes committed:
fix typo on workshop title

sebacodes committed:
add setp 1 to the workshop

jdwilkin4 reviewed (commented):
This is off to a good start. left some comments 👍🏾

sebacodes committed:
fix: typos, workshop order, intro and step 1 hints and test

sebacodes committed:
fix: typo on index.md file

sebacodes committed:
Restore intro.json file to previous version

sebacodes committed:
fix: changed workshop order within the original restored file

sebacodes committed:
fix: added missing linkElement to test 2 on step 1

sebacodes commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68eaeaf8b317a4294f170f40.md:
````diff
@@ -0,0 +1,56 @@
+Your `link` tag should have an `href` atribute
+
+```js
+const hrefAttribute = linkElement.getAttribute('href');
````
created

sebacodes committed:
add step 2 to the workshop

sebacodes committed:
add step 3 to the workshop

sebacodes committed:
fix: added css to the step 2

sebacodes committed:
fix: added css to the step 2

sebacodes committed:
fix: added css to the step 3

sebacodes committed:
add step 4 to the workshop

sebacodes committed:
add step 5 to the workshop, and fix explanation on step 4

sebacodes committed:
add step 6 to the workshop

sebacodes committed:
add step 7 to the workshop

sebacodes committed:
add step 8 to the workshop

sebacodes committed:
add step 9 to the workshop

sebacodes committed:
add step 10 to the workshop

sebacodes committed:
add step 11 to the workshop

sebacodes committed:
add step 12 to the workshop

sebacodes committed:
add step 13 to the workshop

sebacodes committed:
add step 14 to the workshop

sebacodes committed:
add step 15 to the workshop and fix description on step 14

sebacodes committed:
add step 16 to the workshop and fix css property on step 15

sebacodes committed:
add step 17 to the workshop and fix css value on step 16

sebacodes committed:
add step 18 to the workshop and add missing classes to the html file on step 15, 16 and 17

sebacodes committed:
add step 19 to the workshop and add fix order value  on step 15, 16, 17 and 18

sebacodes committed:
add solution on step 19

jdwilkin4 commented on :
@sebacodes 

Make sure to use `git status` often before pushing changes up. That will help you check if unwanted files are about to be committed. 👍🏾

jdwilkin4 reviewed (commented):
It looks like your `curriculum/structure/blocks/workshop-colorful-boxes.json` is not up to date. It currently lists 14 steps but there are 19 step files. 

Before you work on anything else, you will need to resolve that. If you need help, ask the cohort to help you resolve this issue. Otherwise, when you try to run it locally you get this error
<img width="1261" height="339" alt="Screenshot 2025-10-15 at 4 08 01 PM" src="https://github.com/user-attachments/assets/a552ec03-9f98-4ff5-9e2e-14208b4a0a92" />

jdwilkin4 committed:
fix: remove extra file

jdwilkin4 commented on client/i18n/locales/english/intro.json:
```diff
@@ -6956,6 +6908,12 @@
+        "intro": [
+          "In this workshop, you will practice working with CSS flexbox by designing a set of colored boxes."
+        ]
+      },
```
This looks like it is in the right place because it is the second flexbox workshop

<img width="752" height="251" alt="Screenshot 2025-10-16 at 7 52 33 AM" src="https://github.com/user-attachments/assets/682761d6-a2f9-4205-91f7-0b5d8fb3ab6b" />

jdwilkin4 committed:
fix: mismatch for step files

jdwilkin4 committed:
fix: incorrect placement of workshop in intro.json file

jdwilkin4 committed:
fix: incorrect intro name

jdwilkin4 committed:
fix: missing final solution

jdwilkin4 reviewed (commented):
I pushed up a few comments to resolve the following issues:

- removed the extra file that showed your git log
- moved the intro to the correct spot. It was under the new responsive web design cert instead of the normal full stack one
- fixed the name for the workshop in the intro.json file
- added the missing final solution. You added part of it but not all so it wasn't showing in the preview
- added the missing mongo ids to the workshop file so they are in sync. 

I tested things locally and it is now working on my end. 

You will need to pul down these changes into your local environment and resolve the other comments regarding the step language that ilenia gave you.

sebacodes commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ed661e9f463c155bb677a1.md:
```diff
@@ -0,0 +1,59 @@
+
+# --description--
+
+Now, add the title "CSS flexbox colorful boxes" in a `<h1>` tag inside the `header` element.
```
fixed

sebacodes committed:
fix: typos, format and explanation on step 1 to 4

sebacodes committed:
fix: typos, explanation, grammar, missing div and wrong alignment property

sebacodes committed:
fix: added missing instructions for the workshop on step 3

sebacodes committed:
fix: added missing html code in step 4 to 19

sebacodes committed:
add hints and test to step 4 to 19

sebacodes commented:
Step 3 became a bit overloaded with instructions, hints, and tests. This happened because I forgot to include these instructions:

- Add an `h3` element with the text `Box 1` through `Box 6`.
- Add a `p` element with the text `Red` through `Indigo`.

Can I add new steps?, one for the `h3` elements and another for the `p` elements. This is to made a better structure for the workshop and make it more understandable.

If this is possible. Do I just need to add a new step and then modify the .json file with the appropriate hash and step number?. And also change the number in the header of each .md file?

Thank.

jdwilkin4 commented:
> If this is possible. Do I just need to add a new step and then modify the .json file with the appropriate hash and step number?. And also change the number in the header of each .md file?

You should never have to manually update details like this. That will cause errors like you had before.
It is best to use the commands listed in the docs here 

https://contribute.freecodecamp.org/how-to-work-on-workshops/#using-the-scripts-manually

Make sure to run these commands in the workshop directory. 

hope that helps

sebacodes committed:
added 2 new steps for better workshop structure

sebacodes committed:
fix: add test on step 12

sebacodes committed:
fix: typo on test on step 12

sebacodes committed:
add new workshop structure

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68eaeaf8b317a4294f170f40.md:
````diff
@@ -0,0 +1,57 @@
+const linkElement = document.querySelector('link');
+const hrefAttribute = linkElement.getAttribute('href');
+assert.exists(hrefAttribute, 'style.css');
+```
````
check https://github.com/freeCodeCamp/freeCodeCamp/blob/3f7ad26a00a97f9b0b72f4fbc9727d8aaaa07137/curriculum/challenges/english/blocks/lab-business-card/6690e10ebe2181212abc9652.md#L50-L56

sebacodes committed:
fix: test step 1 to 3, 6, 13 and 19 to 21

a2937 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef134c17c96be0af5de613.md:
````diff
@@ -0,0 +1,106 @@
+Your class `flex-container` should have a property `align-content` with the value `start`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.flex-container')?.align-content, 'start');
````
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.flex-container')?.getPropVal('align-content'), 'start');
```

a2937 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef16591f684df7c1d702e1.md:
```diff
@@ -0,0 +1,105 @@
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Colored Boxes</title>
+    
```
```suggestion
     <link href="styles.css" rel="stylesheet"/>
```

a2937 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1750c8ca08009f518b61.md:
````diff
@@ -0,0 +1,106 @@
+Your class `flex-container` should have a property `align-content` with the value `end`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.flex-container')?.align-content, 'end');
````
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.flex-container')?.getPropVal('align-content'), 'end');
```

a2937 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1750c8ca08009f518b61.md:
```diff
@@ -0,0 +1,106 @@
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Colored Boxes</title>
+    
```
```suggestion
    <link href="styles.css" rel="stylesheet"/>
```

a2937 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1835b7772808b7185d11.md:
````diff
@@ -0,0 +1,106 @@
+Your class `flex-container` should have a property `align-content` with the value `space-around`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.flex-container')?.align-content, 'space-around');
````
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.flex-container')?.getPropVal('align-content'), 'space-around');
```

a2937 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef19492dbaff149c384f6c.md:
```diff
@@ -0,0 +1,106 @@
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Colored Boxes</title>
+    
```
```suggestion
        <link href="styles.css" rel="stylesheet"/>
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef0d6d156f86a6ac1824c6.md:
```diff
@@ -0,0 +1,108 @@
+    </div>
+  </body>
+</html>
+--fcc-editable-region--
```
please make sure the editable region is in one single file, the set up you have here with `--fcc-editable-region--` in both files is invalid

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1750c8ca08009f518b61.md:
editable region needs to be adjusted to be in one single file only in this step 9 too

I will not point it out anymore, please update all steps

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef19492dbaff149c384f6c.md:
````diff
@@ -0,0 +1,106 @@
+You can continue
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.box')?.flex, '0 0 250px');
````
you can't have a test that does not fail on seed code of the step

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1b1c657f1b2396d98592.md:
```diff
@@ -0,0 +1,105 @@
+
+# --hints--
+
+Your class `box` should have a property `flex` replace the first number `0` with the value `1`.
```
this sentence is not grammatically correct

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef2080ba680d4c8c0213aa.md:
```diff
@@ -0,0 +1,105 @@
+
+# --hints--
+
+Your class `box` should have a property `flex` replace the third number `250px` with the value `400px`.
```
this also needs to be adjusted.

majestic-owl448 reviewed (commented):
Please run the tests locally with `FCC_BLOCK=workshop-colorful-boxes pnpm run test:curriculum:content`, there is seed code in various steps that does not pass the tests of the previous step.
Also, never use `we` or `us` (including `let's`)

also, the editable region need to be on only one file, it can't be in both files.

I have left some comments where I found these issues, but please make sure to adjust also in other steps.

a2937 reviewed (commented):
Multiple test files that rely on camper-provided CSS do not have a link to `styles.css`. This means no matter what changes the camper makes, no style information will ever be picked up by the tests. Please fix it.

sebacodes commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef19492dbaff149c384f6c.md:
```diff
@@ -0,0 +1,106 @@
+
+Let's dive into `flex` property, this sets how a flex item will grow or shrink to fit the space available in its container.
+
+In our `styles.css` file, we have defined `flex: 0 0 250px` for the `box` class. The first number represents `flex-grow`, the second represents `flex-shrink`, and the last one represents `flex-basis`. In the next section, we are going to see examples of each value.
```
fixed

sebacodes committed:
fix: editable region, grammar and 4 tests

sebacodes committed:
add changes on step 13

sebacodes committed:
fix: delete editable region on html step 12

sebacodes committed:
added missing steps, fix editable region

sebacodes committed:
fix: deleted extra editable region on step 21

sebacodes committed:
fix: deleted extra editable region on step 26

l3onhard committed:
fix(curriculum): adapt description, hints/tests and seed of step 1

l3onhard committed:
fix(curriculum): adapt seeded head element in steps 2-26 to solution in step 26

l3onhard committed:
fix(curriculum): adapt h1 text in steps 2-26 to solution in step 26

l3onhard committed:
fix(curriculum): test locally and adapt (text and tests) step 2

l3onhard committed:
fix(curriculum): remove unnecessary white space in step 1

l3onhard committed:
fix(curriculum): adapt CSS selector in href test of step 1

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef9ac77f6b4176b773e3ac.md:
````diff
@@ -0,0 +1,141 @@
+Create a class called `box5`.
+
+```js
+assert.exist(new __helpers.CSSHelp(document).getStyle('.box5'));
````
```suggestion
assert.exists(new __helpers.CSSHelp(document).getStyle('.box5'));
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1eda8fdbe33f87cfc71d.md:
````diff
@@ -0,0 +1,106 @@
+
+```html
+--fcc-editable-region--
+<!DOCTYPE html>
````
```suggestion
<!DOCTYPE html>
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1eda8fdbe33f87cfc71d.md:
```diff
@@ -0,0 +1,106 @@
+}
+
+.box {
+  flex: 0 0 250px;
```
```suggestion
  flex: 1 0 250px;
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef2080ba680d4c8c0213aa.md:
```diff
@@ -0,0 +1,103 @@
+}
+
+.box {
+  flex: 0 0 250px;
```
```suggestion
  flex: 1 1 250px;
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef9855a1aa425c78b0bc4a.md:
```diff
@@ -0,0 +1,138 @@
+}
+
+.box {
+  flex: 0 0 250px;
```
```suggestion
  flex: 1 1 400px;
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f46fcc34da869b1005455d.md:
```diff
@@ -0,0 +1,118 @@
+}
+
+.box {
+  flex: 0 0 250px;
```
```suggestion
  flex: 1 1 400px;
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f471d46fa193b002a5bfb8.md:
```diff
@@ -0,0 +1,244 @@
+}
+
+.box {
+  flex: 0 0 250px;
```
```suggestion
  flex: 1 1 400px;
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef0b3237fef991c92d9c19.md:
```diff
@@ -0,0 +1,85 @@
+
+# --description--
+
+Now add a new `div` element with the class `flex-container` to your `header`. Then create six `div` elements with the class `box` inside the `div` you just created.
```
you are asking to put it inside `header`, but it's not in the `header` in the next step. I am leaving comments to make it added below the header.


```suggestion
Now add a new `div` element with the class `flex-container` below your `header`. Then create six `div` elements with the class `box` inside the `div` you just created.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1b1c657f1b2396d98592.md:
```diff
@@ -0,0 +1,106 @@
+}
+--fcc-editable-region--
+.box {
+  flex: 0 0 250px;
```
```suggestion
  flex: 1 1 400px;
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1eda8fdbe33f87cfc71d.md:
```diff
@@ -0,0 +1,106 @@
+
+# --description--
+
+Now change the second number in `flex` from `0` to `1` in the `box` class of our `styles.css` file.
```
```suggestion
Now change the second number in `flex` from `1` to `0` in the `box` class of our `styles.css` file.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1eda8fdbe33f87cfc71d.md:
```diff
@@ -0,0 +1,106 @@
+
+# --hints--
+
+Your class `box` should have a property `flex` replace the second number `0` with the value `1`.
```
```suggestion
Your class `box` should have a property `flex` with a value of `0 0 400px`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1b1c657f1b2396d98592.md:
````diff
@@ -0,0 +1,106 @@
+The `box` class should have a property `flex`, replace the first number `0` with the value `1`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.box')?.getPropVal('flex'), '1 0 250px');
````
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.box')?.getPropVal('flex'), '0 1 400px');
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1eda8fdbe33f87cfc71d.md:
````diff
@@ -0,0 +1,106 @@
+Your class `box` should have a property `flex` replace the second number `0` with the value `1`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.box')?.flex, '0 1 250px');
````
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.box')?.flex, '0 0 400px');
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef2080ba680d4c8c0213aa.md:
```diff
@@ -0,0 +1,103 @@
+
+# --description--
+
+Finally, change the last value in `flex` from `250px` to `400px` in the `box` class of our `styles.css` file.
```
```suggestion
Finally, change the last value in `flex` from `400px` to `250px` in the `box` class of our `styles.css` file.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef2080ba680d4c8c0213aa.md:
```diff
@@ -0,0 +1,103 @@
+
+# --hints--
+
+Your `box` class should have a property `flex` replace the third number `250px` with the value `400px`.
```
```suggestion
Your `box` class should have a property `flex` with a value of `0 0 250px`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef2080ba680d4c8c0213aa.md:
````diff
@@ -0,0 +1,103 @@
+Your `box` class should have a property `flex` replace the third number `250px` with the value `400px`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.box')?.flex, '0 0 400px');
````
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('.box')?.flex, '0 0 250px');
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f471d46fa193b002a5bfb8.md:
```diff
@@ -0,0 +1,244 @@
+    <link rel="stylesheet" href="styles.css" />
+  </head>
+  <body>
+    <h1>Colored Boxes Layout</h1>
```
```suggestion
    <header>
      <h1>Colored Boxes Layout</h1>
    </header>
```

majestic-owl448 reviewed (commented):
I have left comments so that the starting css has `1 1 400px` and they are slowing changed to `0 0 250px`.
Also there isn't an `header` in the prototype, what brought you to add it? But I added some suggestions to make that addition work.

I think the tests should all pass after this round of suggestions is added.

l3onhard committed:
fix(curriculum): apply suggestions from code review

- Add `header` element to seeds and solution
- fix typos
- remove fcc-editable-region
- adapt how the `flex` property of the box class is changed and tested

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

l3onhard committed:
fix(curriculum): test locally and adapt (text and tests) step 3

l3onhard committed:
fix(curriculum): change flex property in steps 3-11 to match step 12

l3onhard committed:
fix(curriculum): test locally and adapt (text and tests) step 4

sebacodes committed:
fix: color and test on steps 15, 17, 19, 21, 23 and 25

- Change background color from HEX to RGB
- Updated tests on step 15 y 19, this fix the problem with the properties with a dash on the name

a2937 committed:
Merge branch 'main' into add-flexbox-workshop

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68eaeaf8b317a4294f170f40.md:
```diff
@@ -0,0 +1,62 @@
+
+In this workshop you will practice working with CSS flexbox by designing a set of colored boxes.
+
+To begin the project, link to the external stylesheet `styles.css` from your `index.html` document.
```
```suggestion
To begin, link your `styles.css` file to the HTML document.
```

sebacodes replied:
fixed

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef0b3237fef991c92d9c19.md:
```diff
@@ -0,0 +1,99 @@
+
+Now add a `div` element with the class `flex-container` below your `header`. Then create six `div` elements with the class `box` inside the `div` you just created.
+
+The `styles.css` file has the `flex-container` and `box` class already populated.
```
Since this is a css workshop, campers need to write the css themselves. 
We only give CSS to campers in situations where the workshop is focused on other teaching aspects like JavaScript, React, TypeScript, etc. 

The goal here is the practice CSS and specifically, flexbox. So none of these styles should be given to them. You will need to add steps for all of these styles.

sebacodes replied:
Ok, i will create the steps to all the classes with the respective properties. 

Can i add a brief explanation of the CSS properties? Since some steps will be really simple.

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef0b3237fef991c92d9c19.md:
````diff
@@ -0,0 +1,99 @@
+assert.lengthOf(divElsInsideContainer, 6);
+```
+
+Inside the `div` with the `flex-container` class, all of your six `div` elements should have the class `box`.
````
```suggestion
All six of your `div` elements inside of the `.flex-container` element should have the class `box`.
```

sebacodes replied:
fixed

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f23761964a40289166e2fb.md:
````diff
@@ -0,0 +1,133 @@
+  border-radius: 5px;
+  order: 0;
+}
+```
````
all of this should be removed for reasons mentioned earlier

jdwilkin4 reviewed (commented):
Started reviewing the first few steps. Will continue reviewing more later on.

But here are some things to fix now

- Remove all instances of first person. I still see language like `"Let's`.  https://contribute.freecodecamp.org/how-to-work-on-coding-challenges/#challenge-descriptionsinstructions
- you will need to remove all places where CSS is provided for them. Remember this is a CSS workshop, and they will need to practice writing it out themselves. So you will need to add in some steps there

sebacodes committed:
fix: enhance explanation on step 1 and 3

sebacodes committed:
add step 3, fix: text on step 13 and 22

sebacodes committed:
fix: implement step 4 review feedback and clarify CSS explanation

sebacodes committed:
add setp 5

sebacodes committed:
add new step 6

sebacodes committed:
add new step 7, fix test on step 5 and 6

sebacodes committed:
add new step 8

sebacodes committed:
add new steps 17 to 29

enhance: create new steps based on review feedback
fix: change instructions to second person based on review feedback

sebacodes committed:
fix: grammar and explanation on workshop

l3onhard commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f91df5776e693049fb8942.md:
````diff
@@ -0,0 +1,66 @@
+}
+```
+
+Center the content of the `h1` element by setting its `text-align` property with the value `center`. Also create a `margin-bottom` property with the value `10px` to add margin between the bottom of the `h1` element and any element that goes bellow.
````
```suggestion
Then center the content of the `h1` element by setting its `text-align` property to `center`. 

Also set the `margin-bottom` property to `10px`. This will push everything below the `h1` element down 10 pixels.
```

l3onhard commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f91df5776e693049fb8942.md:
```diff
@@ -0,0 +1,66 @@
+
+# --hints--
+
+Create a `h1` element
```
```suggestion
You should have an `h1` selector.
```

l3onhard commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f91df5776e693049fb8942.md:
````diff
@@ -0,0 +1,66 @@
+assert.equal(new __helpers.CSSHelp(document).getStyle('h1')?.getPropVal('text-align'), 'center');
+```
+
+Add `margin-bottom` property with the value `10px`
````
```suggestion
Your `h1` selector should have a `margin-bottom` property set to `10px`.
```

l3onhard commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f91df5776e693049fb8942.md:
````diff
@@ -0,0 +1,66 @@
+assert.exists(new __helpers.CSSHelp(document).getStyle('h1'));
+```
+
+Add `text-align` property with the value `center`
````
Could we also write: "Your `h1` CSS rule ..."?
Or even just: "Your `h1` rule ..."?

sebacodes committed:
fix: various corrections to make workshop ready for review

- Apply suggestions on text and tests based on review
- Standardization in test code
- Restore pnpm-lock.yaml file from commit 77dad9d

sebacodes committed:
fix: standardization to workshop style for test language on steps

ilenia-magoni commented on curriculum/structure/blocks/workshop-colorful-boxes.json:
```diff
@@ -0,0 +1,56 @@
+    { "id": "68ef9c7dbdab54899bdf1738", "title": "Step 43" },
+    { "id": "68f471d46fa193b002a5bfb8", "title": "Step 44" }
+  ],
+  "blockType": "workshop",
```
```suggestion
  "blockLabel": "workshop",
```

per https://github.com/freeCodeCamp/freeCodeCamp/pull/62548

sebacodes committed:
Update curriculum/structure/blocks/workshop-colorful-boxes.json

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

sebacodes committed:
fix: standardization to workshop language on steps 40, 42 and 44

Dario-DC commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f91df5776e693049fb8942.md:
````diff
@@ -0,0 +1,68 @@
+assert.equal(new __helpers.CSSHelp(document).getStyle('h1')?.getPropVal('text-align'), 'center');
+```
+
+Your `h1` element should have a `margin-bottom` property set to `10px`.
````
```suggestion
Your `h1` selector should set the `margin-bottom` property to `10px`.
```

Dario-DC commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f92924265e9185f1bb391b.md:
```diff
@@ -0,0 +1,72 @@
+
+# --description--
+
+Now go to the `styles.css` file and create a `flex-container` class. This workshop is focused on the CSS flex properties. To practice these, add a `display` property with the value `flex` to the class created. 
```
```suggestion
Going back to the `styles.css` file, create a `flex-container` class selector and set the `display` property to the value `flex`. 
```

Dario-DC commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f92924265e9185f1bb391b.md:
```diff
@@ -0,0 +1,72 @@
+
+Now go to the `styles.css` file and create a `flex-container` class. This workshop is focused on the CSS flex properties. To practice these, add a `display` property with the value `flex` to the class created. 
+
+The `display` property controls how an element is rendered on the page. When you define it with `flex` value, this enables Flexbox layout mode, which gives you control to arrange and align the items inside the container.
```
```suggestion
The `display` property controls how an element is rendered on the page. When you define it with the `flex` value, this enables Flexbox layout mode, which gives you control to arrange and align the items inside the container.
```

Dario-DC commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f92924265e9185f1bb391b.md:
````diff
@@ -0,0 +1,72 @@
+assert.exists(new __helpers.CSSHelp(document).getStyle('.flex-container'));
+```
+
+You should have a property `display` with the value `flex`.
````
```suggestion
You should set the `display` property to the value `flex` within the `.flex-container` selector.
```

Dario-DC commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f932c803f4d3d9f511b436.md:
```diff
@@ -0,0 +1,70 @@
+
+# --description--
+
+To continue, add a `flex-wrap` property with the value `wrap` to the `flex-container` class.
```
```suggestion
To continue, add a `flex-wrap` property with the value `wrap` to the `.flex-container` class selector.
```

Dario-DC commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f23761964a40289166e2fb.md:
```diff
@@ -0,0 +1,116 @@
+Box 3
+Box 4
+Box 5
+Box 6
```
These must be `h2`s because there's no reason to jump from `h1` to `h3`. All the tests and following steps should be changed accordingly.
```suggestion
Inside each `.box` element, add an `h2` element with the text `Box` followed by a space and a sequential number starting from `1`.
```

Dario-DC commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f23761964a40289166e2fb.md:
````diff
@@ -0,0 +1,116 @@
+```js
+const sixthH3El = document.querySelector("div.box:nth-of-type(6) > h3");
+assert.equal(sixthH3El.textContent, "Box 6");
+```
````
````suggestion
You should have an `h2` element with the text `Box 1` inside the first `div` with the class `box`.

```js
const firstH2El = document.querySelector("div.box:nth-of-type(1) > h2");
assert.equal(firstH2El.textContent, "Box 1");
```

You should have an `h2` element with the text `Box 2` inside the second `div` with the class `box`.

```js
const secondH2El = document.querySelector("div.box:nth-of-type(2) > h2");
assert.equal(secondH2El.textContent, "Box 2");
```

You should have an `h2` element with the text `Box 3` inside the third `div` with the class `box`.

```js
const thirdH2El = document.querySelector("div.box:nth-of-type(3) > h2");
assert.equal(thirdH2El.textContent, "Box 3");
```

You should have an `h2` element with the text `Box 4` inside the fourth `div` with the class `box`.

```js
const fourthH2El = document.querySelector("div.box:nth-of-type(4) > h2");
assert.equal(fourthH2El.textContent, "Box 4");
```

You should have an `h2` element with the text `Box 5` inside the fifth `div` with the class `box`.

```js
const fifthH2El = document.querySelector("div.box:nth-of-type(5) > h2");
assert.equal(fifthH2El.textContent, "Box 5");
```

You should have an `h2` element with the text `Box 6` inside the last `div` with the class `box`.

```js
const sixthH2El = document.querySelector("div.box:nth-of-type(6) > h2");
assert.equal(sixthH2El.textContent, "Box 6");
```
````

Dario-DC commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef0d6d156f86a6ac1824c6.md:
```diff
@@ -0,0 +1,86 @@
+---
+
+# --description--
+
```
I think this is a bit misleading. `align-content` controls the arrangement of items along the cross axis. This is something that should be mentioned explicitly. Also, at this stage it seems `justify-content` is the right property to set in order to actually see something on the page.

sebacodes committed:
Update curriculum/challenges/english/blocks/workshop-colorful-boxes/68f91df5776e693049fb8942.md

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

sebacodes committed:
Update curriculum/challenges/english/blocks/workshop-colorful-boxes/68f91df5776e693049fb8942.md

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

sebacodes committed:
Update curriculum/challenges/english/blocks/workshop-colorful-boxes/68f91df5776e693049fb8942.md

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

sebacodes committed:
Apply suggestions from code review

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

sebacodes commented on curriculum/tsconfig.tsbuildinfo:
Acknowledged

johndoe987654321 committed:
refactor: improve flexbox workshop and code quality

- Include code review suggestions
- Reorganize workshop steps for better understanding
- Remove duplicated properties in CSS file
- Change flex-container height to 600px for better align-content visualization
- Added workshop to client/i18n/locales/english/intro.json
- Added workshop to curriculum/structure/superblocks/full-stack-developer.json
- Delete curriculum/tsconfig.tsbuildinfo file

sebacodes committed:
fix: workshop inclusion and delete file

majestic-owl448 reviewed (commented):
are you able to add the workshop to `curriculum/structure/superblocks/responsive-web-design-v9.json` and the related section in `intro.json`?

sebacodes committed:
fix: add workshop to responsive-web-design-v9 files

sebacodes committed:
fix: add workshop to responsive-web-design-v9 files

majestic-owl448 commented on curriculum/i18n-curriculum:
please do not delete, it needs to be removed from the PR, not deleted

majestic-owl448 commented:
to remove `curriculum/i18n-curriculum` from the PR do this:

```bash
git switch main
git pull upstream main
git switch add-flexbox-workshop
git merge main
git checkout main -- curriculum/i18n-curriculum
git add curriculum/i18n-curriculum
git commit -m "sync curriculum/i18n-curriculum"
git push origin add-flexbox-workshop
```

NOTE: after `git merge main` you could need to solve merge conflicts, you can ask for help about it, but please be careful

sebacodes committed:
Merge branch 'freeCodeCamp:main' into main

majestic-owl448 committed:
Merge branch 'main' of https://github.com/freeCodeCamp/freeCodeCamp into add-flexbox-workshop

majestic-owl448 committed:
intro.json

majestic-owl448 committed:
index.md

majestic-owl448 committed:
remove from full-stack-developer.json

majestic-owl448 committed:
remove client/config/growthbook-features-default.json from PR

sebacodes committed:
Merge branch 'freeCodeCamp:main' into add-flexbox-workshop

sebacodes committed:
refactor: improve flexbox workshop
- Reorganize workshop steps for better understanding
- Change flex property in .box class for better mobile and low resolution screens
- Fix duplicated solution in step 43 and 44
- Fix errors that prevent test to pass on steps 33, 35, 37, 39 and 41

sebacodes committed:
sync curriculum/i18n-curriculum

sebacodes committed:
remove file lecture-introduction-to-python/6839b3295323f563efc68f5c.md from PR

majestic-owl448 commented on curriculum/challenges/english/blocks/lecture-introduction-to-python/6839b3295323f563efc68f5c.md:
you should be able to remove it with these commands:

```bash
git switch main
git pull upstream main
git switch add-flexbox-workshop
git merge main
git checkout main -- curriculum/challenges/english/blocks/lecture-introduction-to-python/6839b3295323f563efc68f5c.md
git commit -m "remove curriculum/challenges/english/blocks/lecture-introduction-to-python/6839b3295323f563efc68f5c.md"

majestic-owl448 commented:
example of issue with rewriting the commit history: this comment is not on the related file anymore https://github.com/freeCodeCamp/freeCodeCamp/pull/62735#discussion_r2527465995 but it's still unresolved

majestic-owl448 reviewed (commented):
to remove the unwanted file
```bash
git switch main
git pull upstream main
git switch add-flexbox-workshop
git merge main
git checkout main -- curriculum/challenges/english/blocks/lecture-introduction-to-python/6839b3295323f563efc68f5c.md
git commit -m "remove curriculum/challenges/english/blocks/lecture-introduction-to-python/6839b3295323f563efc68f5c.md"
```

sebacodes committed:
Merge branch 'main' into add-flexbox-workshop

sebacodes committed:
remove curriculum/challenges/english/blocks/lecture-introduction-to-python/6839b3295323f563efc68f5c.md

sebacodes committed:
refactor: update workshop
- Remove unused property on box class
- Apply review comment on box class properties
- Delete step 13 that create property that is not used
- Delete unused property on steps 13 to 43
- All test pass after running FCC_BLOCK=workshop-colorful-boxes pnpm run test:curriculum:content

majestic-owl448 commented:
when you solve the merge conflict, make sure to put `lab-pricing-plans-layout` after this workshop

sebacodes committed:
fix: resolve merge conflict with main

sebacodes committed:
fix: improve explanation by adding line breaks

sebacodes committed:
Merge branch 'freeCodeCamp:main' into add-flexbox-workshop

sebacodes committed:
Merge branch 'freeCodeCamp:main' into add-flexbox-workshop

sebacodes commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68f91df5776e693049fb8942.md:
````diff
@@ -0,0 +1,68 @@
+ property: value;
+}
+```
+
````
Acknowledged

sebacodes commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68eaeaf8b317a4294f170f40.md:
```diff
@@ -0,0 +1,62 @@
+
+In this workshop you will practice working with CSS flexbox by designing a set of colored boxes.
+
+To begin, link your `styles.css` file to the HTML document.
```
Acknowledged, you are right. It makes much more sense to put this step after.

sebacodes committed:
fix: apply review comments

sebacodes commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1b1c657f1b2396d98592.md:
```diff
@@ -0,0 +1,100 @@
+
+# --description--
+
+Now change the first number (flex-grow) in `flex` from `1` to `0` in the `box` class of our `styles.css` file.
```
Acknowledged. I was about to move another, but this makes more sense.

sebacodes commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1b1c657f1b2396d98592.md:
```diff
@@ -0,0 +1,100 @@
+
+# --description--
+
+Now change the first number (flex-grow) in `flex` from `1` to `0` in the `box` class of our `styles.css` file.
```
Acknowledged, taking the second review comment as definitive. It is better rephrased.

sebacodes committed:
Apply review comments (round 2 - Dec 22)

- Fix grammar in documentation
- Reorder steps as suggested
- Improve steps explanation and instructions

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

sebacodes committed:
Apply review comments II (round 2 - Dec 22)

- Fix grammar in steps
- Update CSS to use proper concepts
- Improve test descriptions

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

sebacodes committed:
fix: apply review comments III (round 2 - Dec 22)

- Move steps to the correct positions
- Fix `.box` selector properties according to step changes
- Fix Step 2 code that prevent test on Step 1 to pass
- All test pass

majestic-owl448 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1b1c657f1b2396d98592.md:
```diff
@@ -0,0 +1,100 @@
+
+# --description--
+
+The `flex-grow` property controls how much extra space the flex item should take up if there is free space available in the container.
```
missing explanation that `flex` is a shorthand that includes `flex-grow`, please explicitly say that `flex` is a shorthand for `flex-grow`, `flex-shrink` and `flex-basis`

sebacodes replied:
Acknowledged, that explanation is in step 11 where the .box selector is first introduced, after display: flex. In previous code reviews the criteria was to avoid repeating concepts already taught. Let me know if that applies to this case too or if I should add a brief explanation.

step 11 [694b46a414bb2f5079868ed1.md]

sebacodes replied:
done!

majestic-owl448 commented on client/src/pages/learn/responsive-web-design-v9/workshop-colorful-boxes/index.md:
can you please delete this file? there is been a new recent change and we are not using them anymore

sebacodes replied:
done!

sebacodes committed:
Apply review comments III (round 3 - Jan 30)

- Fix css property
- delete index.md file that is not needed

sebacodes committed:
fix: add `flex` explanation to the step 29

majestic-owl448 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef1b1c657f1b2396d98592.md:
```diff
@@ -0,0 +1,102 @@
+
+The `flex` property controls the size and behavior of the items inside a flexible container. It is compose by three properties: `flex-grow` (1), `flex-shrink` (1), and `flex-basis` (100px).
+
+Now, getting back to the `.box` selector, change the first number in `flex` (corresponding to the `flex-grow` value) from `1` to `0`.
```
please have the part where it say what changes to make in the code be the last sentence in the instructions

sebacodes committed:
Apply review comments IV (round 4 - Feb 2)
- Fix `flex` property explanation and instructions as suggested in the review

sebacodes reviewed (commented):
All requested changes have been reviewed and applied.

sebacodes committed:
Apply suggestions from code review

Co-authored-by: Jeevankumar S <110320697+Jeevankumar-s@users.noreply.github.com>

majestic-owl448 committed:
Merge branch 'main' into add-flexbox-workshop

majestic-owl448 reviewed (commented):
updated the branch, it was a lot out of sync and making it difficult to review locally

make sure you pull this update before you continue working on this

majestic-owl448 reviewed (commented):
please continue making these changes, make sure the editable region includes only the lines that need to be changed

let me know if you need any help with this, I can come do more suggestions

but it should be the last thing needed before adding the workshop to the curriculum

sebacodes committed:
Apply suggestions from code review II

Co-authored-by: majestic-owl448 <26656284+majestic-owl448@users.noreply.github.com>

sebacodes committed:
Merge branch 'freeCodeCamp:main' into add-flexbox-workshop

sebacodes committed:
fix: restrict editable region to designated code area

- Update the editable region to enable code writing within the specific code block.
- Add missing property to Step 11.

sebacodes committed:
fix: correct workshop block placement in curriculum

sebacodes commented on client/i18n/locales/english/intro.json:
```diff
@@ -5658,6 +5658,12 @@
+        "intro": [
+          "In this workshop, you will practice working with CSS flexbox by designing a set of colored boxes."
+        ]
+      },
```
fixed

majestic-owl448 reviewed (commented):
please check all steps for indentation issues in the seed code, and make sure that when the editable region has no code inside, it has already the indentation for the new code to add

also please use this type of formatting for `body` and `head`, meaning not indented (example

```html
<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8" />
  <title></title>
</head>

<body>
  <main></main>
</body>

</html>
```

let me know if you need more help, I can add more suggestions in the PR

majestic-owl448 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ed661e9f463c155bb677a1.md:
```diff
@@ -0,0 +1,55 @@
+        
+--fcc-editable-region--
+    </body>
+</html>
```
```suggestion
<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Colored Boxes</title>
</head>

<body>
--fcc-editable-region--
  
--fcc-editable-region--
</body>

</html>
```

majestic-owl448 commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/68ef0b3237fef991c92d9c19.md:
```diff
@@ -0,0 +1,73 @@
+
+--fcc-editable-region--
+  </body>
+</html>
```
```suggestion
<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Colored Boxes</title>
  <link rel="stylesheet" href="styles.css" />
</head>

<body>
  <header>
    <h1>Colored Boxes Layout</h1>
  </header>
--fcc-editable-region--
  
--fcc-editable-region--
</body>

</html>
```

sebacodes committed:
fix: correct indentation in HTML and CSS

- Resolve all code review comments regarding indentation.
- Note: Indentation was intentionally skipped for steps that do not require it.
- Fix issues with the editable region in some steps.

sebacodes commented:
> please check all steps for indentation issues in the seed code, and make sure that when the editable region has no code inside, it has already the indentation for the new code to add
> 
> also please use this type of formatting for `body` and `head`, meaning not indented (example
> 
> ```
> <!DOCTYPE html>
> <html lang="en">
> 
> <head>
>   <meta charset="UTF-8" />
>   <title></title>
> </head>
> 
> <body>
>   <main></main>
> </body>
> 
> </html>
> ```
> 
> let me know if you need more help, I can add more suggestions in the PR

All requested changes have been made. I have corrected the HTML and CSS indentation across the project, excluding specific steps where indentation is not required.

**Technical Note on Cursor Behavior:**
I applied the indentation within the workshop steps as planned; however, I noticed a platform behavior constraint regarding the editable regions:

***Initial state:*** The cursor defaults to the beginning of the line instead of the indented position. when a new steps start.

***Workaround:*** If you click into the area or move the cursor with the arrow keys, it snaps to the correct indented position.

sebacodes committed:
Apply suggestions from code review

Co-authored-by: majestic-owl448 <26656284+majestic-owl448@users.noreply.github.com>

Jeevankumar-s commented on curriculum/challenges/english/blocks/workshop-colorful-boxes/692f2c20cb7f9e5050b935a7.md:
```diff
@@ -0,0 +1,83 @@
+  max-height: 120px;
+  color: #000;
+--fcc-editable-region--
+	
```
```suggestion
  
```

sebacodes committed:
Apply suggestions from code review II

Co-authored-by: Jeevankumar S <110320697+Jeevankumar-s@users.noreply.github.com>

sebacodes commented:
Hi @Jeevankumar-s, thanks for the code review. I have a question regarding the indentation.

I made a few changes in the Challenge Editor after running `pnpm run challenge-editor`, and now I'm reviewing all the steps in VS Code. However, I noticed an indentation discrepancy between the two environments.

The Challenge Editor considers a TAB as 2 spaces, while VS Code treats it as 4 spaces. Which standard is the correct one to follow? I noticed this difference in several files, take the next file below as an example:

```
id: 68ef9a0b710ba66dd23ed260
title: Step 38
challengeType: 0
dashedName: step-38
```


Thanks!

sebacodes committed:
fix: correct misplaced CSS properties and code indentation

- Fix misplaced CSS properties introduced during the workshop reorganization
- Fix code indentation as suggested in code review

sebacodes committed:
Merge branch 'freeCodeCamp:main' into add-flexbox-workshop

sebacodes committed:
fix: grammar, wording, and code snippets in the paragraphs from code review

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

Dario-DC reviewed (approved):
Hi @sebacodes. Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

majestic-owl448 reviewed (approved):
LGTM

the PR is in automerge, but is blocked by unresolved conversations, help find unresolved conversations and resolve them to allow the PR to merge
